http-mclient/src/Network/HTTP/MClient.hs

```diff
@@ -138,16 +140,55 @@ send' transactionDecoder decoder request = do
   sendRequest sendRequest' transactionDecoder decoder request
 
 sendRequest
-  :: SendRequest
+  :: forall env a . Env env
+  => SendRequest
   -> TransactionDecoder a
   -> ResponseDecoder a
   -> HTTP.Request
   -> MIO env (Result a)
 sendRequest sendRequest' transactionDecoder decoder request =
-  either transactionDecoder decoder
-    <$> Exception.tryJust selectException (liftIO $ sendRequest' request)
+  either transactionDecoder decoder <$> retriableSendRequest
   where
-    selectException :: HTTP.HttpException -> Maybe HTTP.HttpExceptionContent
-    selectException = \case
-      (HTTP.HttpExceptionRequest _request content) -> pure content
-      _other                                       -> empty
+    retriableSendRequest :: MIO env (Either HTTP.HttpExceptionContent (HTTP.Response LBS.ByteString))
+    retriableSendRequest
+      = Retry.retrying
+        limitedBackoff
+        isRetriable
+      $ const (Exception.tryJust selectException . liftIO $ sendRequest' request)
+      where
+        limitedBackoff :: Retry.RetryPolicy
+        limitedBackoff = Retry.exponentialBackoff 50_000 <> Retry.limitRetries (convertImpure retryLimit)
```

Comment:
Question: @mbj I re-read our comms about this task & in one of them you say we should retry x1. Did you mean that literally? Which would imply a constant single-delay retry policy. I personally prefer the current retry policy.

Reply:
@epicallan I'd want to start with 1x retry for the moment. Literal. We can see in our use cases if these flaky upstreams indeed need more or less.